### PR TITLE
Update relevant packages and components in documentation for OAI 2-4-0

### PR DESCRIPTION
**Pull request by chtyler**
## Description
Updated the documentation to display the latest notebook image package versions and components. 

## How Has This Been Tested?
List was confirmed with QE representative and checked on a test cluster with an OAI 2.4 build. 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: modules/notebook-images-for-data-scientists.adoc [hunk 1]
```diff
@@ -12,7 +12,7 @@ Notebook images are upgraded quarterly to ensure that you are working with the l
 endif::[]
 
 ifndef::upstream[]
-Notebook images are supported for a minimum of one year. Major updates to pre-configured notebook images occur approximately every six months. Therefore, two supported notebook images are typically available at any given time. You can use this support period to update your code to use components from the latest available notebook image.
+Notebook images are supported for a minimum of one year. Major updates to pre-configured notebook images occur approximately every six months. Therefore, multiple supported notebook images are typically available at any given time. You can use this support period to update your code to use components from the latest available notebook image.
```
Comment (grainnejenningsRH) [inline modules/notebook-images-for-data-scientists.adoc:15]: In options-for-notebook-server-environments, you've updated 'approximately' to 'about' and 'notebook images' to 'notebook image versions'. Do we want to do that here as well?

File: modules/options-for-notebook-server-environments.adoc [hunk 2]
```diff
@@ -55,29 +67,53 @@ a| * Python 3.8
...
-
-.2+| PyTorch
-| 2023.1 (Recommended)
-a| * Python 3.9
+.3+| PyTorch
+| 2023.2 (Recommended)
+a| * CUDA 11.8
+* Python 3.9
+* PyTorch 2.0
 * JupyterLab 3.5
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:75]:
```suggestion
* JupyterLab 3.6
```

File: modules/options-for-notebook-server-environments.adoc [hunk 3]
```diff
@@ -87,18 +123,40 @@ a| * Python 3.8
 * Scikit-learn 0.24
 * SciPy 1.6
 
-.2+| Standard Data Science
-| 2023.1 (Recommended)
+.3+| Standard Data Science
+| 2023.2 (Recommended)
 a| * Python 3.9
 * JupyterLab 3.5
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:129]:
```suggestion
* JupyterLab 3.6
```

File: modules/options-for-notebook-server-environments.adoc [hunk 4]
```diff
@@ -112,29 +170,54 @@ a| * Python 3.8
...
 * SciPy 1.6
 
-.2+| TensorFlow
-| 2023.1 (Recommended)
-a| * Python 3.9
+.3+| TensorFlow
+| 2023.2 (Recommended)
+a| * CUDA 11.8
+* Python 3.9
+* JupyterLab 3.5
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:177]:
```suggestion
* JupyterLab 3.6
```

File: modules/options-for-notebook-server-environments.adoc [hunk 5]
```diff
@@ -143,19 +226,42 @@ a| * Python 3.8
 * Scikit-learn 0.24
 * SciPy 1.6
 
-| TrustyAI
-| 
+.2+| TrustyAI
+| 2023.2 (Recommended)
 a| * Python 3.9
 * JupyterLab 3.5
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:232]:
```suggestion
* JupyterLab 3.6
```

**Review by jstourac — CHANGES_REQUESTED** [review]

File: modules/notebook-images-for-data-scientists.adoc (hunk 1 above)
Comment (chtyler) [reply modules/notebook-images-for-data-scientists.adoc:15]: I'll update it here also.

Comment (chtyler) [reply modules/notebook-images-for-data-scientists.adoc]: Updated.

**Comment by chtyler** [comment]
Content to be added: 

Only the latest two notebook image versions are supported (2023.1 and 2023.2). The 1.2 notebook image versions are marked as outdated by a label at 2.4, and are there kept just for the user convenience.

File: modules/options-for-notebook-server-environments.adoc [hunk 6]
```diff
@@ -143,19 +226,42 @@ a| * Python 3.8
...
+* Kfp-tekton 1.5
 * Matplotlib 3.6
 * Numpy 1.24
 * Pandas 1.5
 * Scikit-learn 1.2
 * SciPy 1.10
+* Elyra 3.15
 
 | HabanaAI
 | 2023.1 (Recommended)
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:267]: Will confirm with the RC2 but we expect this to be changed to:
```suggestion
| 2023.2 (Recommended)
```

File: modules/options-for-notebook-server-environments.adoc [hunk 7]
```diff
@@ -143,33 +226,56 @@ a| * Python 3.8
...
+* Codeflare-SDK 0.10
+* Sklearn-onnx 1.15
+* Psycopg 3.1 
+* MySQL Connector/Python 8.0
+
+| 2023.1
+a| * Python 3.9
+* JupyterLab 3.5
+* Notebook 6.5
+* TrustyAI 0.2
```
Comment (jstourac) [inline modules/options-for-notebook-server-environments.adoc:255]:
```suggestion
* TrustyAI 0.3
```

**Review by jstourac — APPROVED** [review]

**Comment by chtyler** [comment]
Merging.

File: modules/options-for-notebook-server-environments.adoc (hunk 5 above)
Comment (jstourac) [reply modules/options-for-notebook-server-environments.adoc:232]: hmm, looks like this was marked as resolved but never was actually incorporated :/

File: modules/options-for-notebook-server-environments.adoc (hunk 2 above)
Comment (jstourac) [reply modules/options-for-notebook-server-environments.adoc:75]: same, resolved but wasn't applied

File: modules/options-for-notebook-server-environments.adoc (hunk 3 above)
Comment (jstourac) [reply modules/options-for-notebook-server-environments.adoc:129]: same

File: modules/options-for-notebook-server-environments.adoc (hunk 4 above)
Comment (jstourac) [reply modules/options-for-notebook-server-environments.adoc:177]: and again :slightly_smiling_face:

**Comment by jstourac** [comment]
Will need to better check the resolved conversation next time :slightly_smiling_face: 

Anyway, will be updated via https://issues.redhat.com/browse/RHODS-12818 hopefully.

**Pull request merged**